### PR TITLE
Use correct user on releases (#17806)

### PR DESCRIPTION
**Pull request by Gusted**
- Backport of #17806

### CODE REVIEW

**Comment by Gusted** [comment]
Ah nice, bots won't update if I just change the base branch.

File: modules/notification/webhook/webhook.go [hunk 1]
```diff
@@ -766,12 +766,12 @@ func sendReleaseHook(doer *models.User, rel *models.Release, action api.HookRele
...
 	}
 
-	mode, _ := models.AccessLevel(rel.Publisher, rel.Repo)
+	mode, _ := models.AccessLevel(doer, rel.Repo)
 	if err := webhook_services.PrepareWebhooks(rel.Repo, models.HookEventRelease, &api.ReleasePayload{
 		Action:     action,
 		Release:    convert.ToRelease(rel),
 		Repository: convert.ToRepo(rel.Repo, mode),
-		Sender:     convert.ToUser(rel.Publisher, nil),
+		Sender:     convert.ToUser(doer, nil),
```
Comment (6543) [inline modules/notification/webhook/webhook.go:774]:
```suggestion
		Sender:     convert.ToUser(doer, doer),
```

Comment (6543) [reply modules/notification/webhook/webhook.go:774]: that should also be fixed on main ...

Comment (Gusted) [reply modules/notification/webhook/webhook.go:774]: #17820

Comment (Gusted) [reply modules/notification/webhook/webhook.go:774]: Wait actually... I had a quick look at the other usages, they all use `nil` and not `doer`.

Comment (6543) [reply modules/notification/webhook/webhook.go:774]: well could call it a bug - @Gusted nil should only be used if doer is not known

**Comment by 6543** [comment]
@Gusted sorry ...

https://github.com/go-gitea/gitea/pull/17820#issuecomment-979657384

TLD: my `nil should only be used if doer is not known` - for rest API

but this is for webhooks :sweat:

**Review by 6543 — APPROVED** [review]

**Review by wxiaoguang — APPROVED** [review]

**Pull request merged**